app/scripts/services/membership/membership.js

```diff
@@ -0,0 +1,150 @@
+'use strict';
+
+angular
+  .module('openshiftConsole')
+  .factory('MembershipService', function($filter) {
+
+    var annotation = $filter('annotation');
+
+    var isLastRole = function(userName, roleBindings) {
+      return _.filter(
+              roleBindings,
+              function(binding) {
+                return _.some(binding.subjects, { name: userName });
+              }).length === 1;
+    };
+
+    // internal helper to disambiguate objects with same name, etc:
+    //  - Role view
+    //  - ClusterRole view
+    // arbitrary key creation via any number of provided arguments
+    // examples:
+    //  - uniqueKey(subject.kind, subject.name) = user-foo
+    //  - uniqueKey(role.namespace, role.name) = proj-1-view
+    var uniqueKey = function() {
+      return _.reduce(
+              _.slice(arguments),
+              function(result, next, i) {
+                return next ?
+                      _.isEqual(i, 0) ?
+                        next :
+                        result + '-' + next :
+                      result;
+              }, '');
+    };
+
+    var getSubjectKinds = function() {
+      return {
+         "User":{
+            "kind":"User",
+            "sortOrder":1,
```

Comment:
The object reduces the need to loop for lookups in `mapRolebindingsForUI()`.  I have sortOrder since order isn't guaranteed for object keys once we render.